02-express-tutorial/app.js

```diff
@@ -1 +1,65 @@
-console.log('Express Tutorial')
+console.log("Express Tutorial");
+
+const express = require(`express`);
+// const path = require(`path`);
+const app = express();
+const { products } = require("./data");
+
+app.use(express.static("./public"));
+
+app.get(`/api/v1/test`, (req, res) => {
+  return res.json({ message: "It worked!" });
+});
+
+app.get(`/api/v1/products`, (req, res) => {
+  return res.json({ products });
+});
+
+app.get(`/api/v1/products/sortPriceLow`, (req, res) => {
+  const lowProducts = products.sort((a, b) => a.price - b.price);
+  return res.json({ lowProducts });
+}); //sort products with prices in ascending order
+
+app.get(`/api/v1/products/sortPriceHigh`, (req, res) => {
+  const highProducts = products.sort((a, b) => b.price - a.price);
+  return res.json({ highProducts });
+}); //sort products with prices in descending order
+
+app.get(`/api/v1/products/:productID`, (req, res) => {
+  const idToFind = parseInt(req.params.productID);
+  const product = products.find((p) => p.id === idToFind);
+  if (!product) {
+    return res.status(404).json({ message: "That product was not found." });
+  } else {
+    return res.json({ product });
+  }
+});
+
+app.get(`/api/v1/query`, (req, res) => {
+  const { search, limit } = req.query;
+  let sortedProducts = [...products];
```

Comment:
Here’s where you’d implement the sort above

Reply:
Okay, I have made a new push trying to implement the changes, but I am having a hard time getting it to work within the query-get. I've got the current syntax because I tried using sort=price and sort=-price inside the if statements but couldn't work out using the "=" and "-" symbols, as putting the whole thing inside a string didn't work either. I'd message you on Slack but I just realized I am not sure of your full name :) I am working on this weeks homework for the time being.

Reply:
Ahhh makes sense. It’s a good implementation to try but don’t spend too much effort on it - the way you’d do it is with string parsing to check if the first character in a string is a “-“. The = should just be in the query string parm, not show up in the parsed parameters